### PR TITLE
Use unique pid file for osquery per launcher run

### PR DESCRIPTION
**Pull request by RebeccaMahany**
Closes https://github.com/kolide/launcher/issues/1599 (hopefully)

### CODE REVIEW

File: pkg/osquery/runtime/runner.go [hunk 1]
```diff
@@ -630,6 +630,24 @@ func (r *Runner) launchOsqueryInstance() error {
...
+			"exiting errgroup",
+			"errgroup", "cleanup PID file",
+		)
+
+		<-o.doneCtx.Done()
+		if err := os.Remove(paths.pidfilePath); err != nil {
+			r.slogger.Log(ctx, slog.LevelInfo,
+				"could not remove PID file",
+				"pid_file", paths.pidfilePath,
+				"err", "err",
```
[the remark this answers is not in the document]
Comment (RebeccaMahany) [reply pkg/osquery/runtime/runner.go:645]: 🤦‍♀️ thank you

**Review by zackattack01 — APPROVED** [review]
looks good!

**Review by James-Pickett — APPROVED** [review]

File: pkg/osquery/runtime/runtime_test.go [hunk 2]
```diff
@@ -103,7 +103,7 @@ func TestCalculateOsqueryPaths(t *testing.T) {
 func TestCreateOsqueryCommand(t *testing.T) {
 	t.Parallel()
 	paths := &osqueryFilePaths{
-		pidfilePath:           "/foo/bar/osquery.pid",
+		pidfilePath:           "/foo/bar/osquery-abcd.pid",
```
Comment (directionless) [inline pkg/osquery/runtime/runtime_test.go:106]: ORLY?

Comment (directionless) [reply pkg/osquery/runtime/runtime_test.go:106]: Becca points out this is a test. So I retract me eyebrow

**Review by directionless — APPROVED** [review]

**Pull request merged**